jax/core.py

```diff
@@ -213,27 +210,18 @@ def new_jaxpr_eqn(invars, outvars, primitive, params, effects, source_info=None)
   return JaxprEqn(invars, outvars, primitive, params, effects, source_info)
 
 
-@total_ordering
 class Var:
   # TODO(frostig,mattjj): We don't override __eq__ or __hash__, so comparison is
   # by object id, but pretty printing might collide.
-  count: int
   suffix: str
   aval: AbstractValue
 
-  def __init__(self, count: int, suffix: str, aval: AbstractValue):
-    self.count = count
+  def __init__(self, suffix: str, aval: AbstractValue):
     self.suffix = suffix
     self.aval = raise_to_shaped(aval)
 
-  def __lt__(self, other):
-    if not isinstance(other, Var):
-      return NotImplemented
-    else:
-      return (self.count, self.suffix) < (other.count, other.suffix)
-
   def __repr__(self):
-    return _encode_digits_alphabetic(self.count) + self.suffix
+    return f'Var(id={id(self)}){self.suffix}:{self.aval.str_short()}'
```

Comment:
```suggestion
    suffix_fmt = f', suffix={self.suffix}' if self.suffix else ''
    return f'Var(id={id(self)}{suffix_fmt}):{self.aval.str_short()}'
```

I could see something like `Var(id=..., suffix=...)` being less surprising for nontrivial suffixes. One suffix we use in our cond internals is `_` and currently that would appear as `Var(id=...)_:aval`.

This is a minor comment and I leave it up to you.

Reply:
Oh great idea, will do.